tour_model_eval/viz_bin_cluster.py

```diff
@@ -15,108 +15,86 @@
-
-    if labels:
-        for i, point in enumerate(points):
-            start_lat = point[1]
-            start_lon = point[0]
-            end_lat = point[3]
-            end_lon = point[2]
-            layer = folium.PolyLine([[start_lat, start_lon],
-                                     [end_lat, end_lon]], weight=2, color=color_map(labels[i]))
-            layer.add_to(map)
-    map.add_child(color_map)
-    return map
 
-
-def specific_bin_map(bins, trips):
-    color_map = cm.linear.Set1_07.to_step(len(bins), index=[i for i in range(len(bins))])
-    # Plot trips in the same bin,
-    map = folium.Map(
-        location=[trips[0].data.start_loc["coordinates"][1], trips[0].data.start_loc["coordinates"][0]],
-        zoom_start=12, max_zoom=30, control_scale=True)
-
-    # Here we to choose a specific bin
-    for t in range(0, 3):
-        for i in range(len(bins[t])):
-            curr_trip = trips[bins[t][i]]
-            layer = folium.PolyLine(
-                [[curr_trip.data.start_loc["coordinates"][1], curr_trip.data.start_loc["coordinates"][0]],
-                 [curr_trip.data.end_loc["coordinates"][1], curr_trip.data.end_loc["coordinates"][0]]], weight=2,
-                color=color_map(t))
-
-            layer.add_to(map)
-    map.add_child(color_map)
-    return map
-
-
-def specific_cluster_map(labels, points, clusters):
+    # The function takes four parameters:
+    # - labels: trip label list after running featurization/pipeline.cluster
+    # - labels_list: the list of selected label
+    # - points: coordinate points from featurization/pipeline.cluster
+    # - clusters: number of clusters
+    map = folium.Map(location=[points[0][1], points[0][0]], zoom_start=12, max_zoom=30, control_scale=True)
     labels_clt = list(set(sorted(labels)))
-    color_map = cm.linear.Set1_07.to_step(clusters, index=[i for i in labels_clt])
-
-    # Plot all clusters with different colors on the map
-    # Choose the first start point to locate the map
-    map = folium.Map(location=[points[0][1], points[0][0]], zoom_start=12, max_zoom=30,
-                             control_scale=True)
-
-    if labels:
-        for i, point in enumerate(points):
-            # Here we can choose a specific cluster to plot
-            if labels[i] == 1:
+    color_map = cm.linear.Set1_07.to_step(clusters + 1, index=[i for i in range(len(labels_clt) + 1)])
+
+    if labels_list:
+        for label in labels_list:
+            for i, point in enumerate(points):
+                if labels[i] == label:
+                    start_lat = point[1]
+                    start_lon = point[0]
+                    end_lat = point[3]
+                    end_lon = point[2]
+                    layer = folium.PolyLine([[start_lat, start_lon],
+                                             [end_lat, end_lon]], weight=2, color=color_map(labels[i]+1))
+                    layer.add_to(map)
+    else:
```

Comment:
this is essentially the same as the two function solution before, with the part in for loop being almost identical.
are the labels a list? can't you filter the selected label list similar to the bins above?

Reply:
`labels`is a list like`[1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 2,
 2,
 2,
 2,
 2,
 3,
 3,
 3,
 3,
 3,
 2,
 2,
 2,
 2,
 1,
 1,
 1,
 1,
 5,
 5,
 5,
 5,
 6,
 6,
 6,
 6,
 0,
 0,
 0,
 0,
 2,
 2,
 2,
 2,
 4,
 4,
 4,
 1,
 1,
 1,
 7,
 7,
 7]`, I will need to compare the label for each trip with the selected labels, and only plot the points have selected labels. For general clusters, I just need to plot all the points. I don't see the possibility to reduce the code here, unless create another function for the same part of code.
Also, I need to pass in the labels to determine the color map.

Reply:
why can't your `if` then be

```
if sel_labels is None or trip.label in sel_labels:
    # plot point
```